pkg/agent/kubernetes.go

```diff
@@ -94,17 +97,166 @@ func patchNodeStatus(cli *k8sclient.Clientset, fields map[string]string) error {
 	return err
 }
 
-// watchConfigMap watches changes in a ConfigMap object
-func watchConfigMap(cli *k8sclient.Clientset, ns string, name string) (watch.Interface, error) {
-	listOpts := meta_v1.ListOptions{
-		FieldSelector: "metadata.name=" + name,
+// watch is a wrapper around the k8s watch.Interface
+type watch struct {
+	parent  *watcher
```

Comment:
nit/question: do we need the whole parent instance here? Would just the k8sclient be enough? Would make the architecture prolly slightly cleaner.

Reply:
It is mostly there so we can log using the watcher's logger from the 'bottom part' of the watcher, the watch. I could have passed over it just as a log.Logger but I don't see the point.